azurerm/resource_arm_private_endpoint.go

```diff
@@ -0,0 +1,380 @@
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmPrivateEndpointPrivateLinkServiceConnection(input []interface{}) *[]network.PrivateLinkServiceConnection {
+	results := make([]network.PrivateLinkServiceConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateLinkServiceID := v["private_link_service_id"].(string)
+		groupIds := v["group_ids"].([]interface{})
+		requestMessage := v["request_message"].(string)
+		name := v["name"].(string)
+
+		result := network.PrivateLinkServiceConnection{
+			Name: utils.String(name),
+			PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+				GroupIds:             utils.ExpandStringSlice(groupIds),
+				PrivateLinkServiceID: utils.String(privateLinkServiceID),
+				RequestMessage:       utils.String(requestMessage),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func flattenArmPrivateEndpointPrivateLinkServiceConnection(input *[]network.PrivateLinkServiceConnection) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		v := make(map[string]interface{})
+
+		if name := item.Name; name != nil {
+			v["name"] = *name
+		}
+		if privateLinkServiceConnectionProperties := item.PrivateLinkServiceConnectionProperties; privateLinkServiceConnectionProperties != nil {
+			v["group_ids"] = utils.FlattenStringSlice(privateLinkServiceConnectionProperties.GroupIds)
+			v["private_link_service_connection_state"] = flattenArmPrivateLinkServiceConnectionState(privateLinkServiceConnectionProperties.PrivateLinkServiceConnectionState)
+			if privateLinkServiceId := privateLinkServiceConnectionProperties.PrivateLinkServiceID; privateLinkServiceId != nil {
+				v["private_link_service_id"] = *privateLinkServiceId
+			}
+			if requestMessage := privateLinkServiceConnectionProperties.RequestMessage; requestMessage != nil {
+				v["request_message"] = *requestMessage
+			}
+		}
+
+		results = append(results, v)
+	}
+
+	return results
+}
+
+func flattenArmPrivateEndpointInterface(input *[]network.Interface) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	results := make([]string, 0)
+
+	for _, item := range *input {
+		if id := item.ID; id != nil {
+			results = append(results, *id)
+		}
+	}
+
+	return []interface{}{results}
+}
+
+func flattenArmPrivateLinkServiceConnectionState(input *network.PrivateLinkServiceConnectionState) []interface{} {
+	if input == nil {
+		return make([]interface{}, 0)
+	}
+
+	result := make(map[string]interface{})
+
+	if actionRequired := input.ActionRequired; actionRequired != nil {
```

Comment:
Not a blocker but i find this is easier to review and read 
```suggestion
	if v := input.ActionRequired; v != nil {
```

Reply:
Removed attribute